### PR TITLE
Introduce mbedtls_ecp_write_key_ext

### LINKED ISSUE #8835: mbedtls_ecp_write_key is hard to use
`mbedtls_ecp_write_key` takes a buffer and size as arguments and fills the whole buffer, padding the key with zeros. Depending on whether the key is on a Weierstrass curve or a Montgomery curve, the padding is either at the beginning or at the end of the buffer. This makes the function hard to use, especially since this is not documented.

The goal of this issue is:

* Document the current behavior.
* Add tests for the current behavior. At the moment, we only test `mbedtls_ecp_write_key` with a buffer of the exact expected length, in which case there is no difficulty.
* Document, implement and test a new function that writes from the beginning of the buffer and returns the number of bytes that have been written.
* Deprecate `mbedtls_ecp_write_key`. (Or just replace it by the new function if we do this task as part of a major release.)

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
Introduce `mbedtls_ecp_write_key_ext()` with sane handling of the output buffer length. Deprecate `mbedtls_ecp_write_key()`. Resolves #8835.

## PR checklist

- [x] **changelog** provided
- [x] **backport** no (new feature)
- [x] **tests** provided

### CODE REVIEW

File: include/mbedtls/ecp.h [hunk 1]
```diff
@@ -1370,6 +1370,26 @@ int mbedtls_ecp_read_key(mbedtls_ecp_group_id grp_id, mbedtls_ecp_keypair *key,
...
+ *                  where `grp->nbits` is the private key size in bits.
+ * \param buf       The output buffer for containing the binary representation
+ *                  of the key.
+ * \param buflen    The total length of the buffer in bytes.
+ *                  #MBEDTLS_ECP_MAX_BYTES is always sufficient.
+ *
+ * \return          \c 0 on success.
+ * \return          #MBEDTLS_ERR_ECP_BUFFER_TOO_SMALL if the \p key
+ *                  representation is larger than the available space in \p buf.
+ * \return          Another negative error code on different kinds of failure.
```
Comment (mpg) [inline include/mbedtls/ecp.h:1395]: Should probably mention `BAD_INPUT_DATA` if the private key is not set.

**Review by mpg — APPROVED** [review]
LGTM, only a minor suggestion about the documentation.

**Comment by gilles-peskine-arm** [comment]
I've rebased on top of the merge of #8845.

**Comment by gilles-peskine-arm** [comment]
Another PR was merged that added a call to the deprecated function. I fixed that, and also noticed that I'd forgotten to make the `key` parameter `const` on the new function.

**Review by mpg — DISMISSED** [review]
LGTM!

**Review by minosgalanakis — COMMENTED** [review]
LGTM but could we resolve the conflict and clear CI before merging?

**Review by minosgalanakis — APPROVED** [review]
LGTM

**Review by paul-elliott-arm — APPROVED** [review]
LGTM (Based on Manuel's previous approval + 1 merge commit)

**Pull request merged**